src/main/webapp/MarkerInfoWindow.js

```diff
@@ -19,52 +19,174 @@ class MarkerInfoWindow {
+      titleInput.id = MarkerInfoWindow.TITLE_INPUT;
+      titleInput.placeholder = MarkerInfoWindow.DEFAULT_TITLE;
+
+      let bodyInput = document.createElement("textarea");
+      bodyInput.id = MarkerInfoWindow.BODY_INPUT;
+      bodyInput.placeholder = MarkerInfoWindow.DEFAULT_BODY;
+
+      // if this marker not new, populate the form with the current information
+      if (this.myMarker.isEditing()) {
+        if (this.myMarker.title) {
+          titleInput.innerHTML = this.myMarker.title;
+        }
+        if (this.myMarker.body) {
+          bodyInput.innerHTML = this.myMarker.body;
+        }
+      }
+
+      let submitBtn = document.createElement("button");
+      submitBtn.id = MarkerInfoWindow.SUBMIT_BTN;
+      submitBtn.innerHTML = "Enter";
+
+      form.appendChild(titleInput);
+      form.appendChild(bodyInput);
+      form.appendChild(submitBtn);
+      return form;
+    }
+
+    /**
+     * Puts together the title and body of the info window
+     */
+    makeRightColumn() {
+      let rightCol = document.createElement("div");
+      rightCol.classList.add(MarkerInfoWindow.RIGHT_COLUMN);
+      if (this.myMarker.isEditing()) {
+        rightCol.appendChild(this.makeContentForm());
+      } else {
+        rightCol.appendChild(this.makeTitle());
+        rightCol.appendChild(this.makeBody());
+      }
+      return rightCol;
+    }
+
+    /**
+     * Returns the html string to be displayed in this info window
+     */
+    makeContent() {
+      let contentWrapper = document.createElement("div");
+      contentWrapper.classList.add(MarkerInfoWindow.CONTENT_WRAPPER);
+      contentWrapper.appendChild(this.makeLeftColumn());
+      contentWrapper.appendChild(this.makeRightColumn());
+      let result = contentWrapper.outerHTML;
+      contentWrapper.remove();
+      return result;
     }
 
     /** Closes the displayed info window */
     close() {
       this.googleInfoWindow.close();
+      this.googleInfoWindow.setContent(null);
+    }
+
+    /** Sets the click event for the information window buttons*/
+    setEvents() {
+      let deleteBtn = document.getElementById(MarkerInfoWindow.DELETE_ID);
+      deleteBtn.onclick = () => myMap.deletePermMarker(this.myMarker);
+
+      let editBtn = document.getElementById(MarkerInfoWindow.EDIT_ID);
+      editBtn.onclick = () => {
+        this.myMarker.editing = true;
+        this.open(this.myMarker);
+      };
+
+      let submitBtn = document.getElementById(MarkerInfoWindow.SUBMIT_BTN);
+      if (submitBtn) {
+        submitBtn.addEventListener('click', (ev) => {
+          ev.preventDefault();
+          this.updateContent();
+        });
+      }
+    }
+
+    /** Displays the info window on the map */
+    open(myMarker) {
+      this.close();
+      this.myMarker = myMarker;
+      this.googleInfoWindow.setContent(this.makeContent());
+      this.googleInfoWindow.open(myMap.googleMap, this.myMarker.googleMarker);
     }
 
     /**
```

Comment:
nit: Constructors are usually defined before other methods

Reply:
In this context, the constructor calls functions that need to be defined beforehand so I decided to put it at the very end rather than in the middle.

Reply:
You should be able to call methods in a class, even if they're defined after. Is that not the case?